modules/aerodyn/src/AeroDyn_IO.f90

```diff
@@ -1499,8 +1539,130 @@ SUBROUTINE Cleanup()
+   TYPE(AD_BladePropsType),  INTENT(INOUT)   :: BladeKInputFileData                 ! Data for Blade K stored in the module's input file
+   INTEGER(IntKi),           INTENT(  OUT)   :: ErrStat
+   CHARACTER(*),             INTENT(  OUT)   :: ErrMsg
+      
+   INTEGER(IntKi)                            :: NumCols                             ! number of columns in the file
+   CHARACTER(NWTC_SizeOfNumWord)             :: Words(size(Indx))
+   INTEGER                                   :: IOS(size(Indx))
+   INTEGER                                   :: c                                   ! column index
+   CHARACTER(*), PARAMETER                   :: RoutineName = 'ConvertLineToCols'
+
+      
+   ErrStat = ErrID_None
+   ErrMsg  = ""
+   
+!   CALL GetWords ( Line, Words, size(Indx), NumCols )
+   !IF (MAXVAL(Indx) > NumCols) THEN
+   !   CALL SetErrStat(ErrID_Fatal, "Required column is not available in the table on row "//trim(num2lstr(i))//".", ErrStat, ErrMsg, RoutineName)
+   !   RETURN
+   !END IF
+   
+   ! Read "words" as character strings:
+   NumCols = MAXVAL(Indx)
+   READ(Line, *, IOStat=IOS(1)) Words(1:NumCols)
+      
+   IOS = 0 ! initialize in case we don't read all of the columns
+      
+   ! Note: See order of variable AvailableChanNames in subroutine GetInputColumnIndex() for these variables indices
+   ! Also, we have checked that Indx is non zero and less than MaxCols for each of the required words
+   c=Indx( 1); READ( Words(c), *, IOStat=IOS(c) ) BladeKInputFileData%BlSpn(I)
+   c=Indx( 2); READ( Words(c), *, IOStat=IOS(c) ) BladeKInputFileData%BlCrvAC(I)
+   c=Indx( 3); READ( Words(c), *, IOStat=IOS(c) ) BladeKInputFileData%BlSwpAC(I)
+   c=Indx( 5); READ( Words(c), *, IOStat=IOS(c) ) BladeKInputFileData%BlTwist(I)
+   c=Indx( 6); READ( Words(c), *, IOStat=IOS(c) ) BladeKInputFileData%BlChord(I)
+   c=Indx( 7); READ( Words(c), *, IOStat=IOS(c) ) BladeKInputFileData%BlAFID(I)
+
+   c=Indx(4)
+   IF (c > 0) THEN
+      READ( Words(c), *, IOStat=IOS(c) ) BladeKInputFileData%BlCrvAng(I)
+   END IF
+
+   c=Indx(8)
+   IF (c > 0) THEN
+      READ( Words(c), *, IOStat=IOS(c) ) BladeKInputFileData%BlCb(I)
+   END IF
+
+   c=Indx(9)
+   IF (c > 0) THEN
+      READ( Words(c), *, IOStat=IOS(c) ) BladeKInputFileData%BlCenBn(I)
+   END IF
+
+   c=Indx(10)
+   IF (c > 0) THEN
+      READ( Words(c), *, IOStat=IOS(c) ) BladeKInputFileData%BlCenBt(I)
+   END IF
+
+   IF (ANY(IOS /= 0)) THEN
+      CALL SetErrStat(ErrID_Fatal, "Unable to read numeric data from all columns in the table on row "//trim(num2lstr(i))//".", ErrStat, ErrMsg, RoutineName)
+      RETURN
+   END IF
+      
+END SUBROUTINE ConvertLineToCols
+!----------------------------------------------------------------------------------------------------------------------------------
+SUBROUTINE GetInputColumnIndex(HeaderLine, Indx, ErrStat, ErrMsg)
```

Comment:
In the future, it'll be great to move these to NWTC_IO

Reply:
I moved `GetInputColumnIndex` to NWTC_IO, but it's a little more challenging to get `ConvertLineToCols` to be generic enough to move there. I'll save that for someone else later. :)